x-pack/plugins/observability/server/assistant_functions/get_slo_change_point.ts

```diff
@@ -0,0 +1,297 @@
+        sortDirection: 'desc',
+        page: '1',
+        perPage: '25',
+      };
+      console.log('start', start);
+      console.log('end', end);
+      console.log('shortWindow', shortWindow);
+      console.log('longWindow', longWindow);
+      console.log('alertStart', alertStart);
+      const slos = await sloClient.find.execute(params);
+      let absoluteStartTime: string | undefined;
+
+      if (alertStart && longWindow) {
+        absoluteStartTime = datemath
+          .parse(`now-${longWindow}`, { forceNow: new Date(alertStart) })
+          ?.toISOString();
+      }
+      const esClient = (await resources.context.core).elasticsearch.client.asCurrentUser;
+
+      const mappedItems = slos.results.map((item): SLOSourceItem => {
+        return {
+          name: item.name,
+          sourceIndex: item.indicator.params.index,
+        };
+      });
+      const index = mappedItems[0].sourceIndex;
+      let dataView: DataView;
+
+      const dataViewId = (await dataViewsClient.getIdsWithTitle()).find((id) => {
+        return id.title === index;
+      })?.id;
+      if (dataViewId) {
+        dataView = await dataViewsClient.get(dataViewId);
+      } else {
+        dataView = await dataViewsClient.create({
+          title: mappedItems[0].sourceIndex,
+        });
+      }
+
+      const metricFields = dataView.fields
+        .getByType('number')
+        .filter((f) => f.aggregatable === true)
+        .map((f) => f.name);
+
+      const results = await Promise.all(
+        metricFields.map((fieldName) =>
+          getChangePointForField({
+            index: mappedItems[0].sourceIndex,
+            field: fieldName,
+            esClient,
+            start: start || absoluteStartTime || 'now-24h',
+            end: end || 'now',
+            aggregationType: metricOperation,
+          })
+        )
+      );
+
+      const changes = results.map((result) => ({
+        changedAtApproximateTime: result.aggregations?.change_point_request?.bucket?.key,
+        typeOfChange: result.aggregations?.change_point_request?.type,
+        fieldName: result.fieldName,
+      }));
+
+      console.log('changes', changes);
+
+      return {
+        content: {
+          changes,
+          observedPeriodStart: start || absoluteStartTime || 'now-24h',
+          observedPeriodEnd: end || 'now',
+          metricOperation,
+        },
+        data: {
+          dataView,
+        },
+      };
+    }
+  );
+}
+
+async function getChangePointForField({
+  index,
+  field,
+  esClient,
+  start,
+  end,
+  aggregationType,
+}: {
+  index: string;
+  field: string;
+  esClient: ElasticsearchClient;
+  start: string;
+  end: string;
+  aggregationType: 'avg' | 'min' | 'max' | 'sum';
+}) {
+  return new Promise((resolve) => {
+    esClient
+      .search({
+        index,
+        aggregations: {
+          over_time: {
+            auto_date_histogram: {
+              field: '@timestamp',
+              buckets: 40,
```

Comment:
I wasn't sure how to best determine buckets. I'm chatting with Dima about how the change point chart does it so I can match that logic.